lib/ansible/plugins/httpapi/checkpoint.py

```diff
@@ -0,0 +1,132 @@
+short_description: HttpApi Plugin for Checkpoint devices
+description:
+  - This HttpApi plugin provides methods to connect to Checkpoint
+    devices over a HTTP(S)-based api.
+version_added: "2.8"
+"""
+
+import json
+import os
+import re
+
+from ansible.module_utils.basic import to_text
+from ansible.errors import AnsibleConnectionFailure
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+from ansible.plugins.httpapi import HttpApiBase
+from urllib3 import encode_multipart_formdata
+from urllib3.fields import RequestField
+from ansible.module_utils.connection import ConnectionError
+from ansible.utils.display import Display
+
+BASE_HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json'
+}
+
+display = Display()
+
+
+class HttpApi(HttpApiBase):
+    def __init__(self, connection):
+        super(HttpApi, self).__init__(connection)
+        self._auth = None
+
+    def login(self, username, password):
+        if username and password:
+            payload = {'user': username, 'password': password}
+            url = '/web_api/login'
+            response, response_data = self._send_auth_request(
+                url, json.dumps(payload), method='POST', headers=BASE_HEADERS
+            )
+            response = self._response_to_json(self._get_response_value(response_data))
+        else:
+            raise AnsibleConnectionFailure('Username and password are required for login')
+
+        try:
+            self._auth = response['sid']
+        except KeyError:
+            raise ConnectionError(
+                'Server returned response without token info during connection authentication: %s' % response)
+
+    def logout(self):
+        auth_payload = {
+            'grant_type': 'revoke_token',
+            'access_token': self.access_token,
+            'token_to_revoke': self.refresh_token
+        }
+
+        url = self._get_api_token_path()
+
+        response, dummy = self._send_auth_request(url, json.dumps(auth_payload), method='post',
+                                                  headers=BASE_HEADERS)
+
+        self.refresh_token = None
+        self.access_token = None
+
+    def _send_auth_request(self, path, data, **kwargs):
+        try:
+            return self.connection.send(path, data, **kwargs)
+        except HTTPError as e:
+            # HttpApi connection does not read the error response from HTTPError, so we do it here and wrap it up in
+            # ConnectionError, so the actual error message is displayed to the user.
+            error_msg = self._response_to_json(to_text(e.read()))
+            raise ConnectionError('Server returned an error during authentication request: %s' % error_msg)
+
+    def send_request(self, path, body_params):
+        data = json.dumps(body_params) if body_params else '{}'
+        headers = {'Content-Type': 'application/json', 'X-chkp-sid': self._auth}
```

Comment:
```suggestion
        headers = {'Content-Type': 'application/json'}
```
`httpapi` will automatically merge `connection._auth` with the headers you provide.